src/backend/controllers/battleController.js

```diff
@@ -128,6 +128,126 @@ exports.prepareMatch = async (req: Request, res: Response) => {
 	}
 }
 
+exports.prepareMatch3v3 = async (req: Request, res: Response) => {
+	// Checks that we have received the correct body from the frontend 
+	const errors = validationResult(req);
+
+	if (!errors.isEmpty()) {
+		// Return 400 for a bad request
+		return res
+			.status(400)
+			.json({ errors: errors.array() });
+	}
+
+	const { personBeingChallengedId, challengerTankIds } = req.body;
+
+
+	try {
+		// Holds the array of each tank
+		const tankTeamOne = [];
+		const tankTeamTwo = [];
+	
+		const personBeingChallenged = await User.findById(personBeingChallengedId);
+
+		if (personBeingChallenged == null) {
+			console.log('Could not find personBeingChallenged in DB');
+			return res
+				.status()
+				.json({ msg: 'Could not find user being challenged'});
+		} else if (personBeingChallenged.wager == 0) { // Check that they have a wager set
+			console.log('personBeingChallenged does not have a wager set')
+			return res
+				.status(400)
+				.json({ msg: 'personBeingChallenged does not have a wager set'});
+		}
+
+
+		// Get the tank objects of the personBeingChallenged
+		for (const tankId of personBeingChallenged.favoriteTanks) {
+			const tank = await Tank.findById(tankId, 'tankName components casusCode');
+
+			if (tank == null){
+				return res
+					.status(404)
+					.json({ msg: 'Could not find one of the tanks of personBeingChallenged'});
+			}
+			tankTeamOne.push(tank);
+		}
+		// Get the tank objects of the challenger
+		for (const tankId of challengerTankIds) {
+			const tank = await Tank.findById(tankId, 'tankName components casusCode');
+
+			if (tank == null) {
+				return res
+					.status(404)
+					.json({ msg: 'Could not find one of the tanks of the challenger'});
+			}
+			tankTeamTwo.push(tank);
+		}
+
+		// Shouldnt need to check if null since we queried this earlier
+		// Also we need to do this so we can query the challenger user doc
+		const challengerId = await Tank.findById(challengerTankIds[0], 'userId');
```

Comment:
you need to check to make sure that there are exactly 3 tanks in each list. If they give you a list of 0 tanks, it looks like this code would crash

Reply:
I guess if this is handled in the route, then this is a nonissue

Reply:
Yeah it's tested in the route, I did a similar route check for the 3v3Favorites APIs and they worked for sizes 0, 1, and 2

Reply:
(worked as in they got flagged and stopped)